lib/hooks/useFetchAPI.js

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+
+export function useFetchAPI(url, option) {
+  const [data, setData] = React.useState([]);
+  const [loading, setLoading] = React.useState(false);
+  const [errMsg, setErrMsg] = React.useState("");
+
+  React.useEffect(() => {
+    const controller = new AbortController();
+    const { signal } = controller;
+    setLoading(true);
+    async function getDataAPI() {
+      try {
+        if (!url) return;
+        const respon = await fetch(url, option, signal);
+        const data = await respon.json();
+        setData(data);
+        setLoading(false);
+      } catch (error) {
+        setErrMsg(error);
+      }
+    }
+    getDataAPI();
+    return () => controller.abort();
+  }, [url]);
```

Comment:
Codacy has a fix for the issue: [React Hook React.useEffect has a missing dependency: 'option'. Either include it or remove the dependency array.](https://app.codacy.com/gh/devntv/useHooks_Apply_projects/pullRequest?prid=9710817)
```suggestion
  }, [option, url]);
```